volume/store/store.go

```diff
@@ -309,16 +309,20 @@ func (s *VolumeStore) create(name, driverName string, opts, labels map[string]st
 // This makes sure there are no races between checking for the existence of a volume and adding a reference for it
 func (s *VolumeStore) GetWithRef(name, driverName, ref string) (volume.Volume, error) {
 	name = normaliseVolumeName(name)
```

Comment:
I'm thinking it would be better if there's a separate version of Lock called LockWithError() that checks for error. Callers that use this version should always use CancelWithError() to set errors. This would be cleaner. What do you think?